Source: cherry picked from svn
Upstream: yes (r11796)
Reason: configure failure using a 3.X kernel

commit 0419e51bab274b4d32f337f12d3a497bd7c670d9
Author: bart <bart@a5019735-40e9-0310-863c-91ae7b9d1cf9>
Date:   Sun Jun 5 08:51:47 2011 +0000

    Make Valgrind build on Linux kernel 3.0 and beyond. Closes #274926. An official
    statement that the Linux kernel 3.0 API and ABI is compatible with Linux kernel
    2.6 can be found here: http://lkml.org/lkml/2011/5/29/204.
    
    
    git-svn-id: svn://svn.valgrind.org/valgrind/trunk@11796 a5019735-40e9-0310-863c-91ae7b9d1cf9

diff --git a/configure.in b/configure.in
index e95b4d7..559c1b4 100644
--- a/configure.in
+++ b/configure.in
@@ -225,7 +225,7 @@ case "${host_os}" in
         kernel=`uname -r`
 
         case "${kernel}" in
-             2.6.*) 
+             2.6.*|3.*) 
         	    AC_MSG_RESULT([2.6 family (${kernel})])
         	    AC_DEFINE([KERNEL_2_6], 1, [Define to 1 if you're using Linux 2.6.x])
         	    ;;
